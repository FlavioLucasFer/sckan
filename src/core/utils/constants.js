const MAIN_COLOR = '#1ac3d6';
const SECUNDARY_COLOR = '#245257';

export {
    MAIN_COLOR,
    SECUNDARY_COLOR,
};