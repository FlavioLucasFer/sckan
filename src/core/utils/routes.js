const HOME_ROUTE = '/';
const LOGIN_ROUTE = '/login';
const COMPANY_ROUTE = '/companies';
const COMPANY_FORM_ROUTE = `${COMPANY_ROUTE}/form`;
const PROJECT_ROUTE = '/projects';
const PROJECT_FORM_ROUTE = `${PROJECT_ROUTE}/form`;
const SPRINT_ROUTE = '/sprints';
const SPRINT_FORM_ROUTE = `${SPRINT_ROUTE}/form`;
const TASK_ROUTE = '/tasks';
const TASK_FORM_ROUTE = `${TASK_ROUTE}/form`;
const USER_ROUTE = '/users';
const USER_FORM_ROUTE = `${USER_ROUTE}/form`;
const USER_PROFILE = '/user';
const USER_SITTINGS = `${USER_PROFILE}/settings`;

export {
	HOME_ROUTE,
  LOGIN_ROUTE,
  COMPANY_ROUTE,
	COMPANY_FORM_ROUTE,
  PROJECT_ROUTE,
	PROJECT_FORM_ROUTE,
  SPRINT_ROUTE,
	SPRINT_FORM_ROUTE,
  TASK_ROUTE,
	TASK_FORM_ROUTE,
  USER_ROUTE,
	USER_FORM_ROUTE,
	USER_PROFILE,
	USER_SITTINGS,
};