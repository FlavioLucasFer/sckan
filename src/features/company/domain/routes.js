'use strict';

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/companies`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	REMOVE: `${BASE_URL}/remove`,
};

export default ROUTES;
