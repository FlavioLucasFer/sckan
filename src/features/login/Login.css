.app-name {
	font-size: 120px; 
	font-weight: bold;
	font-style: italic;
	margin: 0;
}

.left-col {
	display: flex;
	justify-content: center;
	padding-top: 100px !important;	
	height: 100%;
}

.right-col {
	height: 100%;
	padding: 10px !important;
	padding-top: 100px !important;
}

.signin-button {
	display: flex;
	align-items: center;
	justify-content: center;

	width: 100%;
	font-weight: bold;
	font-size: large;
	height: 50px;
}

@media (max-width: 600px) {
	.left-col {
		height: 50%;
		padding-top: 5px !important;
	}

	.right-col {
		height: 50%;
		padding: 5px !important;
	}
};