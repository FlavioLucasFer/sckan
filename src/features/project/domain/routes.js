'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/projects`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
};

export default ROUTES;
