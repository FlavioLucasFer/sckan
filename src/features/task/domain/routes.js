'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/tasks`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
};

export default ROUTES;
