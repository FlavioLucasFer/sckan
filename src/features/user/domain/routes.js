'use strict'

const { REACT_APP_API_URL } = process.env;

const BASE_URL = `${REACT_APP_API_URL}/users`;

const ROUTES = {
	CREATE: `${BASE_URL}/create`,
	EDIT: `${BASE_URL}/edit`,
	REMOVE: `${BASE_URL}/remove`,
	FIND: `${BASE_URL}/find`,
	FIND_BY_ID: `${BASE_URL}/find/:id`,
	SIGN_IN: `${BASE_URL}/sign-in`,
	LOGOUT: `${BASE_URL}/logout`,
};

export default ROUTES;
