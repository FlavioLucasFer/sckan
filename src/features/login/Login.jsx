import React from 'react';
import { Row, Col, CardPanel, TextInput } from 'react-materialize';
import { useHistory } from 'react-router-dom';

import useFormInput from 'core/hooks/useFormInput';
import { TASK_ROUTE } from 'core/utils/routes';
import { MainRow } from 'core/components/StyledComponents';
import { AppNameH1, LeftMainCol, RightMainCol, SignInButton } from './components/StyledComponents';

function Login() {
	const userIndentification = useFormInput();
	const userPassword = useFormInput();
  const { push } = useHistory();

  function handleSignInClick() {
    push(TASK_ROUTE);
  }

	return (
		<MainRow id='main-row'
			node='main'>
			<LeftMainCol id='left-main-col' 
				node='section'
				s={12} 
				m={6}
				l={7} >
				<AppNameH1 data-testid="app-name">Sckan</AppNameH1>
			</LeftMainCol>

			<RightMainCol id='right-main-col'
				node='section'
				s={12}
				m={6}
				l={5}>
				<Row id='login-card-row'
					style={{ display: 'flex', justifyContent: 'center' }}>
					<Col id='login-card-col' 
						style={{ margin: 0 }} 
						s={12}
						l={11}
						xl={8}>
						<CardPanel id='login-card'>
							<TextInput {...userIndentification}
								id='user-identification-input'
								data-testid='user-identification-input'
								label='User identification'
								placeholder='Type your email or your identification'
								s={12}
								validate />

							<TextInput {...userPassword}
								id='user-password-input'
								data-testid='user-password-input'
								label='Password'
								placeholder='Type your password'
								s={12}
								password
								validate />
								
							<SignInButton id='login-button'
								data-testid='login-button'
								large
								waves="light"
                onClick={handleSignInClick} >
									Sign-in
							</SignInButton>
						</CardPanel>
					</Col>
				</Row>
			</RightMainCol>
		</MainRow>
	);
}

export default Login;