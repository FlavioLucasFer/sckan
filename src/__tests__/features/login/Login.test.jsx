import React from 'react';
import { shallow } from 'enzyme';

import Login from 'features/login/Login';
import { HOME_ROUTE } from 'core/utils/routes';

const mockHistory = {
	push: jest.fn(),
};

jest.mock('react-router-dom', () => ({
	...(jest.requireActual('react-router-dom')),
	useHistory: () => (mockHistory),
}));

describe('Testing Login screen component', () => {
	it('should render correctly', () => {
		const wrapper = shallow(
			<Login />
		);

		expect(wrapper.debug()).toMatchSnapshot();
	});

	it('should navigato to home screen when click on sign-in button', () => {
		const wrapper = shallow(
			<Login />
		);

		wrapper.find({ id: 'signin-button' }).simulate('click');

		expect(mockHistory.push).toBeCalledWith(HOME_ROUTE);
	});
});