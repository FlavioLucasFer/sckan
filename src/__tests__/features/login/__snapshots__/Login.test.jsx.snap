// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Login screen component should render correctly 1`] = `
"<section id=\\"main-row\\" className=\\"row\\" style={{...}}>
  <div id=\\"left-col\\" className=\\"col s12 m6 l7 left-col\\">
    <h1 className=\\"app-name\\" data-testid=\\"app-name\\">
      Sckan
    </h1>
  </div>
  <div id=\\"right-col\\" className=\\"col s12 m6 l5 teal right-col\\">
    <div className=\\"row\\" id=\\"login-card-row\\" style={{...}}>
      <div className=\\"col s12 l11 xl8\\" id=\\"login-card-col\\" style={{...}}>
        <CardPanel id=\\"login-card\\">
          <TextInput value=\\"\\" onChange={[Function: handleChange]} id=\\"user-identification-input\\" data-testid=\\"user-identification-input\\" title=\\"User identification\\" className=\\"col s12\\" validate={true} mandatory={true} required={true} />
          <TextInput value=\\"\\" onChange={[Function: handleChange]} id=\\"user-password-input\\" data-testid=\\"user-password-input\\" title=\\"Password\\" className=\\"col s12\\" password={true} validate={true} mandatory={true} required={true} />
          <Button id=\\"signin-button\\" title=\\"Sign-in\\" className=\\"cyan accent-3 signin-button\\" data-testid=\\"signin-button\\" onClick={[Function: handleSignIn]} />
        </CardPanel>
      </div>
    </div>
  </div>
</section>"
`;
